solutions/task_2.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+"""
+TASK: Create 'Fake' class to run code below without errors.
+"""
+
+import random
+import string
+
+
+class Fake(dict):
+    def __getattr__(self, name):
+        print("Non existing attribute {} has been called".format(name))
+        return self
+
+    def __call__(self, *args, **kwargs):
+        try:
+            dict.__call__(*args, **kwargs)
+            return self
+        except (TypeError, ValueError):
+            print("The attribute has been called with:")
+            if len(args) > 0:
+                print("- positional arguments: {}".format(str(", ".join(args))))
```

Comment:
1) "join" returns string, why do you use "str()" here?
2) I think it's enough: 


        print("- positional arguments: {}".format(args))

Reply:
All standard objects have got `__str__` method, so it allows you pass them to `format `method when you are formatting string